tests/js/basic-tests.js

```diff
@@ -0,0 +1,284 @@
+/* eslint-env node */
+/* Tests for the "pouch" module */
+"use strict";
+var fluid = require("infusion");
+fluid.logObjectRenderChars = 20480;
+var gpii  = fluid.registerNamespace("gpii");
+
+fluid.require("%gpii-couchdb-test-harness");
+gpii.test.couchdb.loadTestingSupport();
+
+var kettle = require("kettle");
+kettle.loadTestingSupport();
+
+require("./lib/checkResponse");
+
+fluid.registerNamespace("gpii.tests.couchdb.basic");
+
+gpii.tests.couchdb.basic.checkRecordAndStartDelete = function (response, body, expectedStatus, expectedBody, deleteRequest) {
+    var record = JSON.parse(body);
+    gpii.test.couchdb.checkResponse(response, body, expectedStatus, expectedBody);
+
+    // DELETE requests must reference a specific revision, as in:
+    // DELETE /recipes/FishStew?rev=1-9c65296036141e575d32ba9c034dd3ee
+    deleteRequest.send({}, { termMap: { rev: record._rev } });
+};
+
+fluid.defaults("gpii.tests.couchdb.basic.caseHolder", {
```

Comment:
Would be nice to have tests for view operations for `gpii.test.couchdb.caseHolder` too.

Reply:
Good point, and done.